src/main/java/org/jabref/model/strings/StringUtil.java

```diff
@@ -177,34 +178,29 @@ public static String getCorrectFileName(String orgName, String defaultExtension)
      * @return the original fileName if fileName.length() <= maxLength. Otherwise, a shortened fileName
      */
     public static String shortenFileName(String fileName, int maxLength) {
-        if (fileName == null || maxLength < 3) {
+        final int ELLIPSIS_LENGTH = ELLIPSIS.length();
+
+        if (fileName == null || maxLength < ELLIPSIS_LENGTH) {
             return "";
         }
 
         if (fileName.length() <= maxLength) {
             return fileName;
         }
 
-        // Reference the ellipsis' length for future usage
-        final int ELLIPSIS_LENGTH = ELLIPSIS.length();
-
-        // Find the index of the extension dot
         int extensionIndex = fileName.lastIndexOf('.');
 
-        // Extract the name and extension part of a file name
         String name;
         String extension;
 
         // Check if file has an extension or not
-        if (0 < extensionIndex && extensionIndex < fileName.length() - 1) {
-            name = fileName.substring(0, extensionIndex);
-            extension = fileName.substring(extensionIndex);
-        } else {
+        extension = FileUtil.getFileExtension(fileName).map(fileExtension -> '.' + fileExtension).orElse("");
+        if (extension.isEmpty()) {
             name = fileName;
-            extension = "";
+        } else {
+            name = fileName.substring(0, extensionIndex);
```

Comment:
Do it differently - use `length() - extension.length()` - and remove line 191 (` int extensionIndex`) --> There should be only one way be followed to determine and handle extensions.